### DISCUSSION TITLE
Converting MOUSE_PARAMS cmd to MOUSE_EVENTS

### DISCUSSION THREAD

**Discussion by WhoAteDaCake** [c-smile/sciter-js-sdk#186] [Q&A] [answered]
Hi, I don't have much experience in C, so I'm a bit confused by how is the value converted to the enum, since I've seen values go higher than `4094`, which is the maximum enum value. Is there some sort of bit shifting going on? Thanks!

**Comment by c-smile** [comment, marked as answer]
cmd can be OR'ed with EVENT_SINKING and EVENT_HANDLED flags.

See: https://sciter.com/developers/for-native-gui-programmers/#events-propagation

**Reply by WhoAteDaCake** [reply]
Thank you so much for that!

**Reply by WhoAteDaCake** [reply]
Is there a need to handle   `MOUSE_HIT_TEST` in a special manner as well? After removing the mask, I am still sometimes getting 4095 as the value

**Reply by c-smile** [reply]
There are two internal mouse related messages:

 MOUSE_HIT_TEST = 0xFFE,
 MOUSE_CHECK    = 0xFFF, // 4095 decimal
 
 MOUSE_HIT_TEST is used for non-rectangular elements - ones that have custom shape for example.
 MOUSE_CHECK is used to setup cursor properly. For example after end of mouse wheel scroll, when mouse was not moved, engine sends this event to shifted content (setup cursor, check for mouse leave/enter, etc.)

Rule of the thumb: do not handle (skip) all messages you are not interested in.
